pallets/parachain-staking/src/lib.rs

```diff
@@ -1034,15 +1051,15 @@ pub mod pallet {
 		/// # </weight>
 		#[pallet::weight(<T as pallet::Config>::WeightInfo::init_leave_candidates(
 			T::MaxCollatorCandidates::get(),
-			T::MaxCollatorCandidates::get() * T::MaxDelegatorsPerCollator::get()
+			T::MaxCollatorCandidates::get().saturating_mul(T::MaxDelegatorsPerCollator::get())
 		))]
 		pub fn init_leave_candidates(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
 			let collator = ensure_signed(origin)?;
 			let mut state = <CollatorState<T>>::get(&collator).ok_or(Error::<T>::CandidateNotFound)?;
 			ensure!(!state.is_leaving(), Error::<T>::AlreadyLeaving);
 			let mut candidates = <CandidatePool<T>>::get();
 			ensure!(
-				candidates.len().saturating_sub(1) as u32 >= T::MinSelectedCandidates::get(),
+				candidates.len().try_into().unwrap_or(u32::MAX) > T::MinRequiredCollators::get(),
```

Comment:
I went with `saturated_into` in f6bba96

Reply:
Ok this is just me being picky AF. Only in case `candidates.len()` us bigger than `u32::MAX` and `T::MinRequiredCollators::get()` is set to `u32::MAX` this is wrong, because it would not fail although it should. That case is quite unlikely, so never mind....